src/persist/src/location.rs

```diff
@@ -325,6 +325,27 @@ pub trait Consensus: std::fmt::Debug {
         expected: Option<SeqNo>,
         new: VersionedData,
     ) -> Result<Result<(), Option<VersionedData>>, ExternalError>;
+
+    /// Return all versions of data stored for this `key` at sequence numbers
+    /// >= `from`.
+    async fn scan(
+        &self,
+        deadline: Instant,
+        key: &str,
+        from: Option<SeqNo>,
+    ) -> Result<Vec<VersionedData>, ExternalError>;
+
+    /// Deletes all historical versions of the data stored at `key` that are < `seqno`,
+    /// iff `seqno` <= the current sequence number.
+    ///
+    /// Returns a recent version and data stored at key iff `seqno` is greater than
+    /// the current sequence number.
+    async fn truncate(
+        &self,
+        deadline: Instant,
+        key: &str,
+        seqno: SeqNo,
+    ) -> Result<Result<(), Option<VersionedData>>, ExternalError>;
```

Comment:
removed the 2 level result - it took me a second to figure out the code structure but its now quite a bit simpler. also, for the tests im just asserting that they error, instead of asserting identical error text across all impls for now